prereq_tools/base.py

```diff
@@ -64,6 +64,151 @@
 else:
     import configparser as ConfigParser
 
+class EnvironmentModule():
+    """Class for utilizing Modules component to load environment modules"""
+
+    def __init__(self):
+        """Load Modules for initializing envirables"""
+        self._module_func = self._init_module_func(lambda *x: False)
+        self._module_load = self._init_mpi_module()
+
+    def set_module_func(self, func):
+        """Set the mpi module func"""
+        self._module_func = func
+
+    @staticmethod
+    def _module_old(*args):
+        """invoke module and save environment"""
+        cmd = ['/usr/bin/modulecmd', 'python']
+
+        for arg in args:
+            cmd.append(arg)
+        output = subprocess.check_output(cmd)
+        exec(output)
+
+    def _init_module_func(self, default_func):
+        """Initialize environment modules"""
+        init_file = None
+        for root, _dirs, files in os.walk('/usr/share/Modules'):
+            for fname in files:
+                if fname == "python.py":
+                    temp = os.path.join(root, fname)
+                    if "init" in temp:
+                        init_file = temp
+                        break
+        if init_file is None:
+            return default_func
+
+        try:
+            subprocess.check_call(['sh', '-l', '-c', 'module -V'],
+                                  stdout=DEVNULL, stderr=DEVNULL)
+        except subprocess.CalledProcessError:
+            return self._module_old
+
+        try:
+            exec(open(init_file).read())
+        except KeyError:
+            return default_func
+
+        # pylint: disable=undefined-variable
+        return module
```

Comment:
what is this module var? a comment would help here

Reply:
it gets defined in the exec call which essentially just prints a bunch of os.environ["..."] = ...